src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -539,20 +540,24 @@ private CompilationTracker GetCompilationTracker(ProjectId projectId)
 
         private Solution AddProject(ProjectId projectId, ProjectState projectState)
```

Comment:
also, global operation can be nested. so, if there is a point for solution re/loading (like KnownUIContext - http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/SolutionBuildMonitor.cs,cd3fac6641d35e3e), you can start global operation there as well without worrying about individual add/remove project case in already loaded solution.